app/code/Magento/Search/Test/Mftf/Section/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled">
+        <annotations>
+            <stories value="Search Term"/>
+            <title value="Mini search field does not appear if suggestions was disabled"/>
+            <description value="Mini search field does not appear if suggestions was disabled"/>
+            <severity value="CRITICAL"/>
+            <testCaseId value="MC-38652"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+
+        <before>
+            <!-- Login as admin -->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <!-- Disable search suggestion -->
+            <magentoCLI command="config:set catalog/search/search_suggestion_enabled 0" stepKey="disableSearchSuggestion"/>
+        </before>
+
+        <after>
+            <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCache">
```

Comment:
Please, do not flush the cache. In 99% cases, this action is not required but it slows down all other tests significantly. Please, clean the proper cache tags only

https://github.com/magento/magento2/blob/e9190a5bb2c4aede2a752e94a3e87dcf9647b604/app/code/Magento/Backend/Test/Mftf/ActionGroup/CliCacheCleanActionGroup.xml#L11

You will not slow down the testing process using this way. 

Thanks

Reply:
Updated my PR